charts/radar-s3-connector/templates/deployment.yaml

```diff
@@ -0,0 +1,133 @@
+              labelSelector:
+                matchExpressions:
+                  - key: "app.kubernetes.io/name"
+                    operator: In
+                    values:
+                    - {{ template "radar-s3-connector.name" . }}
+                  - key: "app.kubernetes.io/instance"
+                    operator: In
+                    values:
+                    - {{ .Release.Name }}
+              topologyKey: "kubernetes.io/hostname"
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+          - name: CONNECT_BOOTSTRAP_SERVERS
+            value: "{{ .Values.kafka }}"
+          - name: CONNECT_REST_PORT
+            value: "8083"
+          - name: CONNECT_GROUP_ID
+            value: "default"
+          - name: CONNECT_CONFIG_STORAGE_TOPIC
+            value: "default.config"
+          - name: CONNECT_OFFSET_STORAGE_TOPIC
+            value: "default.offsets"
+          - name: CONNECT_STATUS_STORAGE_TOPIC
+            value: "default.status"
+          - name: CONNECT_KEY_CONVERTER
+            value: "io.confluent.connect.avro.AvroConverter"
+          - name: CONNECT_VALUE_CONVERTER
+            value: "io.confluent.connect.avro.AvroConverter"
+          - name: CONNECT_KEY_CONVERTER_SCHEMA_REGISTRY_URL
+            value: "{{ .Values.schema_registry }}"
+          - name: CONNECT_VALUE_CONVERTER_SCHEMA_REGISTRY_URL
+            value: "{{ .Values.schema_registry }}"
+          - name: CONNECT_INTERNAL_KEY_CONVERTER
+            value: "org.apache.kafka.connect.json.JsonConverter"
+          - name: CONNECT_INTERNAL_VALUE_CONVERTER
+            value: "org.apache.kafka.connect.json.JsonConverter"
+          - name: CONNECT_OFFSET_STORAGE_FILE_FILENAME
+            value: "/tmp/connect2.offset"
+          - name: CONNECT_REST_ADVERTISED_HOST_NAME
+            value: {{ include "radar-s3-sink-connector.fullname" . }}
+          - name: CONNECT_ZOOKEEPER_CONNECT
+            value: "{{ .Values.zookeeper }}"
+          - name: CONNECTOR_PROPERTY_FILE_PREFIX
+            value: "sink-s3"
+          - name: KAFKA_HEAP_OPTS
+            value: "-Xms256m -Xmx768m"
+          - name: KAFKA_BROKERS
+            value: "{{ .Values.kafka_num_brokers }}"
+          - name: CONNECT_LOG4J_LOGGERS
+            value: "org.reflections=ERROR"
+          - name: AWS_ACCESS_KEY
+            value: "{{ .Values.bucketAccessKey }}"
+          - name: AWS_SECRET_KEY
+            value: "{{ .Values.bucketSecretKey }}"
+          ports:
+            - name: http
+              containerPort: 8083
+              protocol: TCP
+          livenessProbe:
+            exec:
+              command:
+              - /bin/sh
+              - -c
+              - curl  -sf localhost:8083/connectors/radar-s3-sink-connector/status | grep -o '\"state\":\"[^\"]*\"' | tr '\\n' ',' | grep -vq FAILED || exit 1
```

Comment:
If the `/status` could return a 500 HTTP code on when it's not healthy then we wouldn't need this complex logic to check the status

Reply:
I think the API wouldn't 500 when it is not healthy, it would return a 200 and FAILED which is why we do this check. See [Connect-rest-api](https://docs.confluent.io/current/connect/references/restapi.html#get--connectors-(string-name)-status).

Reply:
I see, we haven't wrote that route ourselves